src/storage-controller/src/collection_mgmt.rs

```diff
@@ -553,6 +584,116 @@ where
+                    &introspection_config.storage_collections,
+                    &introspection_config.txns_read,
+                    &introspection_config.persist,
+                )
+                .await;
+
+                let scraper_token = statistics::spawn_statistics_scraper::<
+                    statistics::SourceStatistics,
+                    SourceStatisticsUpdate,
+                    _,
+                >(
+                    self.id.clone(),
+                    // These do a shallow copy.
+                    introspection_config.collection_manager,
+                    Arc::clone(&introspection_config.source_statistics),
+                    prev,
+                    introspection_config.statistics_interval.clone(),
+                    introspection_config.statistics_interval_receiver.clone(),
+                    introspection_config.metrics,
+                );
+                let web_token = statistics::spawn_webhook_statistics_scraper(
+                    introspection_config.source_statistics,
+                    introspection_config.statistics_interval,
+                    introspection_config.statistics_interval_receiver,
+                );
+
+                // Make sure these are dropped when the controller is
+                // dropped, so that the internal task will stop.
+                introspection_config
+                    .introspection_tokens
+                    .lock()
+                    .expect("poisoned")
+                    .insert(self.id, Box::new((scraper_token, web_token)));
+            }
+            IntrospectionType::StorageSinkStatistics => {
+                let prev = snapshot_statistics(
+                    self.id,
+                    introspection_config.recent_upper,
+                    &introspection_config.storage_collections,
+                    &introspection_config.txns_read,
+                    &introspection_config.persist,
+                )
+                .await;
+
+                let scraper_token =
+                    statistics::spawn_statistics_scraper::<_, SinkStatisticsUpdate, _>(
+                        self.id.clone(),
+                        introspection_config.collection_manager,
+                        Arc::clone(&introspection_config.sink_statistics),
+                        prev,
+                        introspection_config.statistics_interval,
+                        introspection_config.statistics_interval_receiver,
+                        introspection_config.metrics,
+                    );
+
+                // Make sure this is dropped when the controller is
+                // dropped, so that the internal task will stop.
+                introspection_config
+                    .introspection_tokens
+                    .lock()
+                    .expect("poisoned")
+                    .insert(self.id, scraper_token);
+            }
+
+            // Truncate compute-maintained collections.
```

Comment:
nit: we don't do anything here, but fine to leave it in to avoid another round of ci

Reply:
I'll fix this in the next one.